ibm/resource_ibm_cis_rate_limit_test.go

```diff
@@ -100,30 +104,35 @@ func testAccCheckIBMCisRateLimitExists(n string, tfRecordID *string) resource.Te
 		}
 
 		tfRecord := *tfRecordID
-		cisClient, err := testAccProvider.Meta().(ClientSession).CisAPI()
+		cisClient, err := testAccProvider.Meta().(ClientSession).CisRLClientSession()
+		if err != nil {
+			return err
+		}
 		recordID, zoneID, cisID, _ := convertTfToCisThreeVar(rs.Primary.ID)
-		foundRecordPtr, err := cisClient.RateLimit().GetRateLimit(cisID, zoneID, recordID)
+		cisClient.Crn = core.StringPtr(cisID)
+		cisClient.ZoneIdentifier = core.StringPtr(zoneID)
+		opt := cisClient.NewGetRateLimitOptions(recordID)
+		foundRecordPtr, _, err := cisClient.GetRateLimit(opt)
 		if err != nil {
 			return err
 		}
 
-		foundRecord := *foundRecordPtr
-		if foundRecord.ID != recordID {
+		foundRecord := foundRecordPtr.Result
+		if *foundRecord.ID != recordID {
 			return fmt.Errorf("Record not found")
 		}
 
-		tfRecord = convertCisToTfThreeVar(foundRecord.ID, zoneID, cisID)
+		tfRecord = convertCisToTfThreeVar(*foundRecord.ID, zoneID, cisID)
 		*tfRecordID = tfRecord
 		return nil
 	}
 }
 
 func testAccCheckIBMCisRateLimitConfigBasic() string {
-	return testAccCheckIBMCisDomainDataSourceConfig_basic1() + fmt.Sprintf(`
-	
+	return testAccCheckIBMCisDomainDataSourceConfigBasic1() + fmt.Sprintf(`
 	resource "ibm_cis_rate_limit" "ratelimit" {
 		cis_id = data.ibm_cis.cis.id
-		domain_id = data.ibm_cis_domain.cis_domain.id
+		domain_id = data.ibm_cis_domain.cis_domain.domain_id
```

Comment:
revert back to `domain_id = data.ibm_cis_domain.cis_domain.id'
 for all existing resource testcases